jvm/src/test/scala/tastyquery/PositionSuite.scala

```diff
@@ -0,0 +1,394 @@
+    )
+  }
+
+  /** Throw and try-catch */
+
+  test("throw") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"ThrowException")
+    assertEquals(collectCode[Throw](tree, code), List("throw new NullPointerException"))
+  }
+
+  test("try") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TryCatch")
+    assertEquals(
+      collectCode[Try](tree, code),
+      List("""try {
+             |      ThrowException().f()
+             |      1
+             |    } catch {
+             |      case _ => 0
+             |    } finally {
+             |      ()
+             |    }""".stripMargin)
+    )
+  }
+
+  /** Types */
+
+  test("typed") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Typed")
+    assertEquals(collectCode[Typed](tree, code), List("1: Int"))
+  }
+
+  test("type-member") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TypeMember")
+    assertEquals(
+      collectCode[TypeMember](tree, code),
+      List(
+        "type TypeMember = Int",
+        "type AbstractType",
+        "type AbstractWithBounds >: Null <: AnyRef",
+        "opaque type Opaque = Int",
+        "opaque type OpaqueWithBounds >: Null <: AnyRef = Null"
+      )
+    )
+  }
+
+  test("type-apply") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TypeApply")
+    assertEquals(collectCode[TypeApply](tree, code), List("Seq[Int]", "A[Int, Seq[String]]"))
+  }
+
+  test("type-ident") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Typed")
+    assertEquals(collectCodeT[TypeIdent](tree, code), List("Int"))
+  }
+
+  test("singleton-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"SingletonType")
+    assertEquals(collectCodeT[SingletonTypeTree](tree, code), List("x.type"))
+  }
+
+  test("refined-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"RefinedType")
+    assertEquals(collectCodeT[RefinedTypeTree](tree, code), Nil)
+  }
+
+  test("by-name-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"ByNameParameter")
+    assertEquals(collectCodeT[ByNameTypeTree](tree, code), List("=> Int"))
+  }
+
+  test("applied-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"AppliedTypeAnnotation")
+    assertEquals(collectCodeT[AppliedTypeTree](tree, code), List("Option[Int]"))
+  }
+
+  test("select-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"SelectType")
+    assertEquals(collectCodeT[SelectTypeTree](tree, code), List("util.Random"))
+  }
+
+  test("annotated-type") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"VarargFunction")
+    assertEquals(collectCodeT[AnnotatedTypeTree](tree, code), List("Int*"))
+  }
+
+  test("match-type".ignore) {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"MatchType")
+    assertEquals(collectCodeT[MatchTypeTree](tree, code), List(""))
+  }
+
+  test("type-tree-bind".ignore) {
```

Comment:
why are these ignored?

Reply:
> https://github.com/scalacenter/tasty-query/blob/7ba4447c45de2316fd36841a121e6f8dde3fb322/shared/src/main/scala/tastyquery/ast/Trees.scala#L95-L97
> 
> Some tests of spans are limited by the method `walkTypeTrees` because it assumes `TypeTree` can only be attached to `Tree`. But there're cases where `TypeTree` is a subtree of another `TypeTree`. For example, `MatchTypeTree` can be a subtree of `TypeLambdaTree`. In such cases, `walkTypeTrees` can't reach all the type trees. But I have to say the specification of the function is accurate.

Because of the limitation of `walkTypeTrees` mentioned above

Reply:
I think it's easy to fix

Reply:
I see, I guess in a separate PR we can define walkTrees/TypeTrees on TypeTree and use that recursively in here